File: GamePlatform/game/BaseGame.java
package GamePlatform.Game;

import javax.swing.JFrame;

public abstract class BaseGame extends JFrame {
    protected String username;
    protected String gameName;
    protected long startTime;
    protected int score;
    
    public BaseGame(String title) {
        super(title);
        this.username = GameSession.getCurrentUser();
        this.gameName = GameSession.getCurrentGame();
        this.startTime = System.currentTimeMillis();
        this.score = 0;
    }
    
    protected void recordScore() {
        int playTime = (int)((System.currentTimeMillis() - startTime) / 1000);
        GameAPI.recordGameScore(gameName, username, score, playTime);
    }
    
    protected void showStats() {
        Map<String, Object> stats = GameAPI.getGameStats(gameName);
        showGameStats(stats);
    }
    
    protected abstract void showGameStats(Map<String, Object> stats);
} 

File: game/base_game.py
from game_api import GameAPI
import time

class BaseGame:
    def __init__(self, game_name):
        self.username = None  # 将从配置文件读取
        self.game_name = game_name
        self.start_time = time.time()
        self.score = 0
        
        # 读取当前用户
        self.load_current_user()
    
    def load_current_user(self):
        try:
            with open('../gameplatform_session.txt', 'r') as f:
                self.username = f.read().strip()
        except:
            print("Warning: Could not load user session")
    
    def record_score(self):
        play_time = int(time.time() - self.start_time)
        GameAPI.record_score(self.game_name, self.username, self.score, play_time)
    
    def show_stats(self):
        stats = GameAPI.get_game_stats(self.game_name)
        self.show_game_stats(stats)
    
    def show_game_stats(self, stats):
        raise NotImplementedError("Subclasses must implement show_game_stats") 